src/sdk/v2/sdk.ts

```diff
@@ -43,44 +44,72 @@ import { InitFhevm } from "../init.js";
 import { PermitV2ParamsValidator } from "./permit.z.js";
 import { EncryptedNumber, EncryptionTypes } from "../types.js";
 
-const initialize = async (params: InitParams & { ignoreErrors?: boolean }) => {
+/**
+ * Initializes the `fhenixsdk` to enable encrypting input data, creating permits / permissions, and decrypting sealed outputs.
+ * Initializes `fhevm` client FHE wasm module and fetches the provided chain's FHE publicKey.
+ * If a valid signer is provided, a `permit/permission` is generated automatically
+ */
+const initialize = async (
+  params: InitParams & { ignoreErrors?: boolean; generatePermit?: boolean },
+): Promise<Result<PermitV2 | undefined>> => {
   // Initialize the fhevm
   await InitFhevm().catch((err: unknown) => {
     if (params.ignoreErrors) {
       return undefined;
     } else {
-      throw new Error(
+      return ResultErr(
         `initialize :: failed to initialize fhenixjs - is the network FHE-enabled? ${err}`,
       );
     }
   });
 
   if (params.provider == null)
-    throw new Error(
+    return ResultErr(
       "initialize :: missing provider - Please provide an AbstractProvider interface",
     );
 
   if (params.securityZones != null && params.securityZones.length === 0)
-    throw new Error(
+    return ResultErr(
       "initialize :: a list of securityZones was provided, but it is empty",
     );
 
   await _store_initialize(params);
+
+  // `generatePermit` must set to `false` to early exit here
+  if (params.generatePermit === false) return ResultOk(undefined);
+
+  // Return the existing active permit
+  const userActivePermit = getPermit();
+  if (userActivePermit.success) return userActivePermit;
+
+  // Create permit and return it
+  return createPermit();
```

Comment:
These changes introduce permit generation to the sdk's `initialize` function. This improves how the sdk interacts with a reactive frontend client. For example, a hook that depends on `const account = wagmi.useAccount()` will automatically re-run when the account changes. If within that hook is `fhenixsdk.initialize(... account ...)` then the sdk will be re-initialized with the new user (via provider / signer), and the correct permit pulled from storage (if it exists already) or created.